### PR TITLE
Improvements to Authentication Flow and Feedbacks

### PR DESCRIPTION
**Pull request by Ememobong28**
**Description:**
This PR aims to improve the overall authentication flow in the application. I have added improvements and features have been added to ensure a smoother user experience and improve data storage based on your past feedbacks.

1.) Enhanced error handling: TechHive now handles scenarios when a user tries to log in without an existing account. This will help to provide clear communication and avoid confusion. 
2.)User Data Storage: Every login activity will now be saved to the database.
3.)UI and Flow Improvements: I made changes to the UI and flow when a user signs up or logs in, making the flow more intuitive. 
4.)Login and Logout: Users can now properly log in and log out of the application. I managed this using React Context. 
5.)Post-Signup Navigation: After a user signs up, they will be directed to the login page to complete the authentication process. 

**Milestones:**
1.) Implement error handling for login attempts with non-existent accounts.
2.) Store each login activity to the database.
3.) Made UI and flow adjustments to the signup and login process.
4.) Implement and test the Login and Logout functionality. 
5.) Redirect users to login page post signup.

**Test Plan:**
1.) Error Handling: Try to log in using a non-existent account. An appropriate error message should appear. Then navigate user to signup page
2.) Data Storage: After logging in, check the database to ensure that the login activity has been stored properly. 
3.) UI and Flow: Navigate through the signup and login process. The flow should be seamless, and any UI changes should be functioning and visually pleasing. 
4.) Login and Logout: Try to log in to the application using a test account. After successful login, try to log out. TechHive should handle both these activities smoothly. 
5.) Post-Signup Navigation: After signing up, the application should redirect to the login page.

**Video Recordings**
1.) How i checked if the users information is being stored when they sign up.
[User Database.mov.zip](https://github.com/Ememobong28/Capstone-Project-TechHive/files/12035329/User.Database.mov.zip)

2.) **Login/Signup Updates**
[Login:Signup Modifications.mov.zip](https://github.com/Ememobong28/Capstone-Project-TechHive/files/12035331/Login.Signup.Modifications.mov.zip)

3.) **Logout Function**
https://github.com/Ememobong28/Capstone-Project-TechHive/assets/84681087/e35258fc-a34b-4d61-afae-1734969864b6

### CODE REVIEW

File: techhive-ui/src/components/Login/Login.jsx [hunk 1]
```diff
@@ -27,34 +29,45 @@ const styles = {
...
+        console.log(response.data);
+        navigate('/');
+      } catch (err) {
+        console.error(err.response.data); 
+        if(err.response.data.message === "User not found"){
+          setError("No account found. Please create an account.");
+          setTimeout(() => navigate('/signup'), 3000);
+        }
+        else {
+          // something else
```
Comment (nicolec2626) [inline techhive-ui/src/components/Login/Login.jsx:60]: are you planning to implement other error messages?

We can commit code with TODO comments for future iterations, but typically we want them to be more informative than this. We also wouldn't want to submit an empty else statement. Instead you can keep just the if-statement and add an informative comment "TODO add additional error handling for X type of errors"

File: techhive-ui/src/components/Login/Login.css [hunk 2]
```diff
@@ -0,0 +1,10 @@
+.alert {
+    color: red;
+    margin-top: 10px;
+    /* Add more styles as needed */
```
Comment (nicolec2626) [inline techhive-ui/src/components/Login/Login.css:4]: we don't need to commit these instructional comments

File: techhive-ui/src/components/Login/Login.jsx [hunk 3]
```diff
@@ -27,34 +29,45 @@ const styles = {
...
-      const response = await axios.post("http://localhost:3000/auth/login", { email, password });
-      console.log(response.data);
-      navigate('/');
-    } catch (err) {
-      console.error(err.response.data); 
-    }
+        const response = await axios.post("http://localhost:3000/auth/login", { email, password });
+        localStorage.setItem('user', JSON.stringify(response.data));
+        setUser(response.data.user);
+        console.log(response.data);
```
Comment (nicolec2626) [inline techhive-ui/src/components/Login/Login.jsx:51]: we don't typically commit console.log statements as they're just used for debugging

File: techhive-ui/src/components/Login/Login.jsx [hunk 4]
```diff
@@ -1,6 +1,8 @@
-import React, { useState } from 'react';
+import React, { useState, useContext } from 'react';
```
Comment (nicolec2626) [inline techhive-ui/src/components/Login/Login.jsx:1]: glad to see you using states and contexts!

**Review by nicolec2626 — COMMENTED** [review]
overall looks good - thanks for incorporating the feedback I provided about user flow/messages when signing up and logging in. also, I am a fan of the confetti

**Comment by nicolec2626** [comment]
Also, great improvement on the PR content. They are looking much better 👍

File: techhive-ui/src/components/Login/Login.jsx (hunk 1 above)
Comment (Ememobong28) [reply techhive-ui/src/components/Login/Login.jsx:60]: Okay noted. Thank you

File: techhive-ui/src/components/Login/Login.css (hunk 2 above)
Comment (Ememobong28) [reply techhive-ui/src/components/Login/Login.css:4]: Okay noted. I thought I took them out. I will go through all my code and take it out with the console logs

**Pull request merged**